modules/couchbase/src/main/java/org/testcontainers/containers/CouchbaseContainer.java

```diff
@@ -0,0 +1,284 @@
+        addFixedExposedPort(8091, 8091);
+        addFixedExposedPort(8092, 8092);
+        addFixedExposedPort(8093, 8093);
+        addFixedExposedPort(8094, 8094);
+        addFixedExposedPort(8095, 8095);
+        addFixedExposedPort(11207, 11207);
+        addFixedExposedPort(11210, 11210);
+        addFixedExposedPort(11211, 11211);
+        addFixedExposedPort(18091, 18091);
+        addFixedExposedPort(18092, 18092);
+        addFixedExposedPort(18093, 18093);
+        setWaitStrategy(new HttpWaitStrategy().forPath("/ui/index.html#/"));
+    }
+
+    public CouchbaseEnvironment getCouchbaseEnvironnement() {
+        if (couchbaseEnvironment == null) {
+            initCluster();
+            couchbaseEnvironment = DefaultCouchbaseEnvironment.builder()
+                    .bootstrapCarrierDirectPort(getMappedPort(11210))
+                    .bootstrapCarrierSslPort(getMappedPort(11207))
+                    .bootstrapHttpDirectPort(getMappedPort(8091))
+                    .bootstrapHttpSslPort(getMappedPort(18091))
+                    .build();
+        }
+        return couchbaseEnvironment;
+    }
+
+    public CouchbaseCluster getCouchbaseCluster() {
+        if (couchbaseCluster == null) {
+            couchbaseCluster = CouchbaseCluster.create(getCouchbaseEnvironnement(), getContainerIpAddress());
+        }
+        return couchbaseCluster;
+    }
+
+    public SELF withClusterUsername(String username) {
+        this.clusterUsername = username;
+        return self();
+    }
+
+    public SELF withClusterPassword(String password) {
+        this.clusterPassword = password;
+        return self();
+    }
+
+    public SELF withMemoryQuota(String memoryQuota) {
+        this.memoryQuota = memoryQuota;
+        return self();
+    }
+
+    public SELF withIndexMemoryQuota(String indexMemoryQuota) {
+        this.indexMemoryQuota = indexMemoryQuota;
+        return self();
+    }
+
+    public SELF withKeyValue(Boolean withKV) {
+        this.keyValue = withKV;
+        return self();
+    }
+
+    public SELF withIndex(Boolean withIndex) {
+        this.index = withIndex;
+        return self();
+    }
+
+    public SELF withQuery(Boolean withQuery) {
+        this.query = withQuery;
+        return self();
+    }
+
+    public SELF withFTS(Boolean withFTS) {
+        this.fts = withFTS;
+        return self();
+    }
+
+    public SELF withTravelSample(Boolean withTravelSample) {
+        this.travelSample = withTravelSample;
+        return self();
+    }
+
+    public SELF withBeerSample(Boolean withBeerSample) {
+        this.beerSample = withBeerSample;
+        return self();
+    }
+
+    public SELF withGamesIMSample(Boolean withGamesIMSample) {
+        this.gamesIMSample = withGamesIMSample;
+        return self();
+    }
+
+    public SELF withNewBucket(BucketSettings bucketSettings) {
+        newBuckets.add(bucketSettings);
+        return self();
+    }
+
+
+    public void initCluster() {
```

Comment:
actually even private would be better. I intend people to use start anyway, Will fix this.